MoreLinq/EndsWith.cs

```diff
@@ -29,21 +29,24 @@ static partial class MoreEnumerable
         /// <typeparam name="T">Type of elements.</typeparam>
         /// <param name="first">The sequence to check.</param>
         /// <param name="second">The sequence to compare to.</param>
+        /// <param name="comparer">Optional equality comparer, defaults to <see cref="EqualityComparer{T}.Default" />.</param>
         /// <returns>
         ///     <c>true</c> if <paramref name="first" /> ends with elements equivalent to <paramref name="second" />.
         /// </returns>
         /// <remarks>
         ///     This is the <see cref="IEnumerable{T}" /> equivalent of <see cref="string.EndsWith(string)" /> and
-        ///     it calls <see cref="object.Equals(object)" /> on pairs of elements at the same index.
+        ///     it calls <see cref="IEqualityComparer{T}.Equals(T,T)" /> on pairs of elements at the same index.
         /// </remarks>
-        public static bool EndsWith<T>(this IEnumerable<T> first, IEnumerable<T> second)
+        public static bool EndsWith<T>(this IEnumerable<T> first, IEnumerable<T> second, IEqualityComparer<T> comparer = null)
```

Comment:
>  is this a convention in this library or in .NET in general?

It's the convention used by LINQ (have a look at [`SequenceEqual`](https://msdn.microsoft.com/en-us/library/system.linq.enumerable.sequenceequal(v=vs.110).aspx), which has overloads, [one where the default comparer will be used](https://msdn.microsoft.com/en-us/library/bb348567(v=vs.110).aspx) and [another where the caller can supply one](https://msdn.microsoft.com/en-us/library/bb342073(v=vs.110).aspx) and .NET in general. Above all, this is not about personal preferences.

An often forgotten idea about LINQ is being able to use expression trees to project a query into another system. An overload without any comparer permits a `first.StartsWith(second)` to be projected into another system whose defaults may be something else or may have a different representation for expressing comparison semantics.